internal/service/event_test.go

```diff
@@ -482,3 +470,59 @@ func TestEventDeviceUpdated(t *testing.T) {
 	require.NoError(err)
 	compareEvents(expectedEvents, events.Items, require)
 }
+
+// =============================== ENROLLMENT REQUEST ========================
+func TestEventEnrollmentRequestApproved(t *testing.T) {
+	require := require.New(t)
+
+	serviceHandler := serviceHandler()
+	testDirPath := t.TempDir()
+	cfg := ca.NewDefault(testDirPath)
+	ca, _, err := fcrypto.EnsureCA(cfg)
+	require.NoError(err)
+	serviceHandler.ca = ca
+	ctx := context.Background()
+
+	// create ER
+	name := uuid.New().String()
+	approval := api.EnrollmentRequestApproval{
+		Approved: true,
+		Labels:   &map[string]string{"label": "value"},
+	}
+	status := api.EnrollmentRequestStatus{}
+	deviceStatus := api.NewDeviceStatus()
+	deviceReadWriter := fileio.NewReadWriter(fileio.WithTestRootDir(t.TempDir()))
+	_, privateKey, _, err := fcrypto.EnsureKey(deviceReadWriter.PathFor("TestCSR"))
+	require.NoError(err)
+	csr, err := fcrypto.MakeCSR(privateKey.(crypto.Signer), name)
+	require.NoError(err)
+	er := api.EnrollmentRequest{
+		ApiVersion: "v1",
+		Kind:       "EnrollmentRequest",
+		Metadata: api.ObjectMeta{
+			Name: lo.ToPtr(name),
+		},
+		Spec: api.EnrollmentRequestSpec{
+			Csr:          string(csr),
+			DeviceStatus: &deviceStatus,
+			Labels:       &map[string]string{"labelKey": "labelValue"}},
+		Status: &status,
+	}
+
+	_, err = serviceHandler.store.EnrollmentRequest().Create(ctx, store.NullOrgId, &er)
+	require.NoError(err)
+	identity := authcommon.Identity{
+		Username: "bar",
+	}
+	ctx = context.WithValue(ctx, authcommon.IdentityCtxKey, &identity)
+	_, stat := serviceHandler.ApproveEnrollmentRequest(ctx, name, approval)
+	expectedEvents := []api.EventReason{
+		api.DeviceContentUpToDate,
+		api.ResourceCreated,
+		api.DeviceApproved,
+	}
+	require.Equal(statusSuccessCode, stat.Code)
+	events, _ := serviceHandler.store.Event().List(context.Background(), store.NullOrgId, store.ListParams{})
+	require.NoError(err)
```

Comment:
I think we are missing the error here

```suggestion
	events, err := serviceHandler.store.Event().List(context.Background(), store.NullOrgId, store.ListParams{})
	require.NoError(err)
```